### ISSUE TITLE
Switch Hunger Games image reporting/flagging to Nutri-Patrol

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/hunger-games#1036]
### Problem
- Nutri-Patrol is now live, with a reporting API, and also a reporting webview that is deeplinkable (2 ways to implement as a result)
- Switch to Nutri-Patrol for image flagging
- Use the classes available in the JS SDK: https://github.com/openfoodfacts/openfoodfacts-js src/nutripatrol.ts, src/schemas/nutripatrol.ts, tests/nutripatrol.spec.ts

### Screenshots
![image](https://github.com/user-attachments/assets/36a22dd9-4bfc-4b6a-b719-23df9ee10fd6)

### Part of
- #37

**Comment by alexfauquette** [comment]
@teolemon Can you link to the API for nutripatrol?

**Comment by teolemon** [comment]
Yes, sorry @alexfauquette : https://nutripatrol.openfoodfacts.org/api/docs

**Comment by teolemon** [comment]
Also, here's the frontend: https://github.com/openfoodfacts/nutripatrol-frontend
and the deployment: https://nutripatrol.openfoodfacts.org/

**Comment by teolemon** [comment]
I came back because of 

![Image](https://github.com/user-attachments/assets/85dfad0f-dee4-4282-860f-75876a48957c)

**Comment by Ramneet04** [comment]
> I'm not sure to get your question. For me a user can report an image, and then they don't care at all. The user report the issue, but don't care about the follow up

Hey, It was regarding the un-flag feature only, check the link I shared, I'm redirecting the users here to flag a product, after the flag it shows "request will get reviewed".
so now if you are on another website, so we can't track if user actually flagged the issue like he really "submitted" it so for that purpose we can't keep track of flagged images! 

> I don't get what is the webview you're talking about. Do you have a link?

I mean to report for a specific product: 
https://nutripatrol.openfoodfacts.org/flag/image/?barcode=0737628064502&image_id=2&source=web&flavor=off

**Comment by alexfauquette** [comment]
Ok, I better understand why it's not possible to unflag images :+1:

For me the goal is to be the less disruptive as possible for the user. It's a side quest.

- The flags should only be visible for connected users
- The UX could either be
	- open the nutripatrol in distinct tab
	- a tooltip that send an API request to flag the image with the selected reason

**Comment by Ramneet04** [comment]
hey I was trying it using OFF Node SDK but I'm getting CORS Error, is it possible to enable CORS on the Nutri-Patrol flagging endpoint so we could submit flags directly from the Hunger Games frontend? or should I stick with the open nutripatrol in distinct tab.

**Comment by teolemon** [comment]
it's been fixed
